### PR TITLE
Updated max page width for better readability on large screens

### PR DESCRIPTION
**Pull request by LayaTaal**
This isn't an open issue, but a few layout things have been bothering me so I wanted to propose this update and see what others think. This is still a work in progress, because it creates some issues with the Explore page that need to be resolved. I will appreciate ideas for that.

What's included:
- Set max page width
- Create vertical margins between page content and header/footer
- End sidebar immediately once content ends, rather than stretching to match the primary content column.

Currently we allow the site to fill the entire width of the browser. This creates a big readability issue on large screens. I also feel like the sidebar doesn't need to stretch all the way to the bottom, but rather can just end when it's content does.

### CODE REVIEW

**Comment by nholden** [comment]
Looking good!

What do you think about allowing the explore page to be full width and only enforcing the max width on other pages? I think we'll end up wanting a full width layout for the incarceration map, too.

**Comment by LayaTaal** [comment]
> Looking good!
> 
> What do you think about allowing the explore page to be full width and only enforcing the max width on other pages? I think we'll end up wanting a full width layout for the incarceration map, too.

That's what I was thinking as well @nholden . Keep those charts nice and large. I'll work on addressing this and ping you once it's ready.

**Comment by LayaTaal** [comment]
@nholden I've updated this with a new Layout component that allows us to let each page decide how wide it should be (i.e. the Explore the Data page should now remain full width. Mind taking a peek for me?

**Comment by nholden** [comment]
Could you talk a little about the difference between the `<Layout>` and `<Primary>` components? Do we need both?

**Comment by LayaTaal** [comment]
> Could you talk a little about the difference between the `<Layout>` and `<Primary>` components? Do we need both?

Hey @nholden I'm just swinging back around to this now. We don't necessarily need a component `<Layout>`, though I do think we need a container of some sort for `<primary>` since there may be a sidebar component.

I was in the middle of removing the the layout component altogether, but then I thought it might be good to keep it separated into a component in case we need to expand it in the future.

Whatcha think?

File: components/Layout.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+// Looks for width prop to be either 'fixed', or 'fluid'
+const Layout = props => {
+  const { width, children } = props;
+  return <StyledDiv width={width}>{children}</StyledDiv>;
+};
+
+export default Layout;
+
+Layout.propTypes = {
+  width: PropTypes.string,
```
Comment (nholden) [inline components/Layout.js]: If we're only supporting two types of widths, could we refactor this to be a boolean? If we need to support more than two types of widths, what do you think about using [`PropTypes.oneOf`](https://reactjs.org/docs/typechecking-with-proptypes.html) to make sure we get a valid value?

File: components/Layout.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  align-items: flex-start;
+  width: 100%;
+  max-width: ${props => (props.width === 'fluid' ? '100%' : props.theme.large)};
+  margin: ${props => (props.width === 'fluid' ? '0' : '4rem auto')};
+
+  @media (min-width: ${props => props.theme.medium}) {
+    flex-flow: row nowrap;
+  }
+
+  @media (min-width: ${props => props.theme.medium}) and (max-width: calc(1028px + 2rem)) {
```
Comment (nholden) [inline components/Layout.js]: Where does the calculation for `max-width` come from? Is there a variable we could use here?

File: components/Sidebar.js [hunk 3]
```diff
@@ -13,29 +14,11 @@ Sidebar.propTypes = {
 };
 
 const StyledAside = styled.aside`
+  flex: 1 0 342px;
```
Comment (nholden) [inline components/Sidebar.js]: Since we're using `342px` in a couple of places, could we extract a variable?

**Pull request merged**